tests/appsec/test_appsec_utils.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+import pytest
+
+from ddtrace import constants
+import ddtrace.appsec.utils as utils
+from tests.utils import override_env
+
+
+@pytest.fixture(autouse=True)
+def reset_template_caches():
+    utils._HTML_BLOCKED_TEMPLATE_CACHE = None
+    utils._JSON_BLOCKED_TEMPLATE_CACHE = None
+    try:
+        yield
+    finally:
+        utils._HTML_BLOCKED_TEMPLATE_CACHE = None
+        utils._JSON_BLOCKED_TEMPLATE_CACHE = None
+
+
+def test_get_blocked_template_no_env_var_html():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML="", DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON="")):
+        assert utils._get_blocked_template("text/html") == constants.APPSEC_BLOCKED_RESPONSE_HTML
+
+
+def test_get_blocked_template_no_env_var_json():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML="", DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON="")):
+        assert utils._get_blocked_template("other") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+        assert utils._get_blocked_template("application/json") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+        assert utils._get_blocked_template("") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+        assert utils._get_blocked_template(None) == constants.APPSEC_BLOCKED_RESPONSE_JSON
+
+
+def test_get_blocked_template_user_file_missing_html():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML="missing.html")):
+        assert utils._get_blocked_template("text/html") == constants.APPSEC_BLOCKED_RESPONSE_HTML
+        assert utils._get_blocked_template("") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+
+
+def test_get_blocked_template_user_file_missing_json():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON="missing.json")):
+        assert utils._get_blocked_template("") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+        assert utils._get_blocked_template("text/html") == constants.APPSEC_BLOCKED_RESPONSE_HTML
+
+
+def test_get_blocked_template_user_file_exists_html():
+    template_path = os.path.join(os.path.dirname(__file__), "blocking_template_html.html")
```

Comment:
`pytest` has a `tmpdir` fixture to help with this:

https://github.com/DataDog/dd-trace-py/blob/97deadb4be6902c07536c4fc0a788b934641cadb/tests/integration/test_integration.py#L93-L106

Reply:
It's not a temporal dir but the dir where the test (and the fixture) is.

Reply:
oh... nvm. I read this too quickly. I saw the open calls, and was thinking we were writing content to a file. Sorry about that.